console/v1/types_console_notification_banner.go

```diff
@@ -0,0 +1,58 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Extension for configuring openshift web console notifications.
+type ConsoleNotificationBanner struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +required
+	Spec ConsoleNotificationBannerSpec `json:"consoleLinksSpec"`
+	// +optional
+	Status ConsoleNotificationBannerStatus `json:"consoleLinksStatus"`
+}
+
+// Represents console notification spec
+type ConsoleNotificationBannerSpec struct {
+	// Text of the notification
+	Text string `json:"text"`
+	// The location of the notification (top, bottom, topBottom)
+	Position BannerPosition `json:"position"`
+	// Object that holds notification link details
+	Link ConsoleLink `json:"link"`
+	// TODO: finalize optional or required
```

Comment:
Per https://github.com/openshift/console/pull/1542/files#r285668581

`+optional` and perhaps following a `theme` conventions like success, danger, warning, info instead of open-ended color options...

Reply:
Can be optional, but we have a requirement to let admins choose specific colors.

Reply:
I'm good with the defaulting.

Tempted to add `SuggestedBackgroundColor` and a small set of reasonable options but I'll refrain :)